src/contracts/GPv2AllowListAuthentication.sol

```diff
@@ -3,28 +3,51 @@ pragma solidity ^0.7.6;
 
 import "@gnosis.pm/util-contracts/contracts/StorageAccessible.sol";
 import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+import "@openzeppelin/contracts/proxy/Proxy.sol";
 import "./interfaces/GPv2Authentication.sol";
-import "./ownership/CustomInitiallyOwnable.sol";
 
 /// @title Gnosis Protocol v2 Access Control Contract
 /// @author Gnosis Developers
-contract GPv2AllowListAuthentication is
-    CustomInitiallyOwnable,
-    GPv2Authentication,
-    StorageAccessible
-{
+contract GPv2AllowListAuthentication is GPv2Authentication, StorageAccessible {
     using EnumerableSet for EnumerableSet.AddressSet;
 
+    address public manager;
+
     EnumerableSet.AddressSet private solvers;
 
-    // solhint-disable-next-line no-empty-blocks
-    constructor(address initialOwner) CustomInitiallyOwnable(initialOwner) {}
+    function initializeManager(address manager_) external {
+        require(manager == address(0), "GPv2: already initialized");
+        manager = manager_;
+    }
+
+    modifier onlyProxyAdmin() {
+        // Slot taken from https://eips.ethereum.org/EIPS/eip-1967#specification
+        // obtained as bytes32(uint256(keccak256('eip1967.proxy.admin')) - 1)
+        bytes32 slot =
+            0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
+        address proxyAdmin;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            proxyAdmin := sload(slot)
+        }
+        require(msg.sender == proxyAdmin, "GPv2: caller not proxyAdmin");
+        _;
+    }
+
+    modifier onlyManager() {
+        require(manager == msg.sender, "GPv2: caller not manager");
+        _;
+    }
+
+    function updateManager(address newManager) external onlyProxyAdmin {
+        manager = newManager;
```

Comment:
If the manager gets set to `address(0)`, then anyone can call `initializeManager` and take ownership of the contract now.

Maybe instead of adding the ability of changing the manager in this PR, we do it as a separate PR. What do you think @fedgiac ?

Reply:
My understanding is that the current changes on the owner are work in progress.
I don't mind if it's done here or done in another PR, but I think that making provision for changing the manager is necessary before releasing.

Reply:
> I think that making provision for changing the manager is necessary before releasing.

Selbstverständlich

Reply:
Reverted the `onlyProxyAdmin`